src/main/java/com/volmit/iris/util/decree/DecreeParameter.java

```diff
@@ -57,7 +56,7 @@ public String getDescription() {
     }
 
     public boolean isRequired() {
-        return param.required();
+        return !hasDefault();
```

Comment:
I changed this for a reason, why change it back? We need default AND required. Yes i know they collide with each other but it doesnt cause any issues, where as this solution, removing required and ASSUMING required if the default isnt there, does not cover all edge cases. 

Revert this and any changes to the Param class too

Reply:
Name me one case where the absence of a default means it's NOT a required parameter
OR
a required parameter has a default value

and I'll do it

Reply:
If its required and has a default value, obviously Iris will still complain if you dont specify it. This case doesnt matter it works either way

However if it is NOT required and does NOT have a default value, NULL is inserted which is SUPER important. I actually use this in a few situations. For example. Iris pregen

/iris pregen [dim]

if dim is specified, parse & use it, otherwise (null, with NO DEFAULT AND NOT REQUIRED) will be used, in which case we can have custom logic to find what world they are currently in and assume that is their context. Im planning on using a commandContext to find out what engine they are talking about.

Basically, the default system is not as good as null checking and being smart about what null actually means. A default system cannot specify "my current location" for example.

Reply:
This makes more sense than I'd like to admit :P

I'll get to it tomorrow, not enough focus to do it right today.